packages/example-generator/src/run.ts

```diff
@@ -0,0 +1,220 @@
+    { cwd: path.dirname(fileURLToPath(import.meta.url)) },
+  );
+  if (mrlConfig == null) {
+    consola.error("Unable to find .monorepolint.config.mjs");
+    process.exit(1);
+  }
+  process.chdir(path.dirname(mrlConfig));
+  const mrlPaths = TEMPLATES.map((template) =>
+    path.join(tmpDir.name, templateExampleId(template), "package.json")
+  );
+  const { stdout: mrlStdout, stderr: mrlStderr } = await promisify(exec)(
+    `pnpm exec monorepolint check --fix --paths ${mrlPaths.join(" ")}`,
+  );
+  consola.log(mrlStdout);
+  consola.log(mrlStderr);
+
+  // Check or copy files
+  if (check) {
+    for (const template of TEMPLATES) {
+      const exampleId = templateExampleId(template);
+      consola.info(`Checking contents of ${exampleId}`);
+      // realpath because globby in .gitignore filter requires symlinks in tmp directory to be resolved
+      const pathLeft = fs.realpathSync(path.join(outputPath, exampleId));
+      const pathRight = fs.realpathSync(path.join(tmpDir.name, exampleId));
+      const compareResult = compareSync(
+        pathLeft,
+        pathRight,
+        {
+          compareContent: true,
+          filterHandler: gitIgnoreFilter(pathLeft, pathRight),
+          excludeFilter: "/.turbo",
+        },
+      );
+      if (!compareResult.same) {
+        consola.error(
+          `Found ${compareResult.differences} differences in ${exampleId} please generate examples again.`,
+        );
+        consola.error(compareResult.diffSet?.filter(d => d.state !== "equal"));
+        process.exit(1);
+      }
+      consola.success(`Contents equal`);
+    }
+  } else {
+    consola.info("Copying generated packages to output directory");
+    for (const template of TEMPLATES) {
+      const exampleId = templateExampleId(template);
+      const exampleOutputPath = path.join(outputPath, exampleId);
+      const exampleTmpPath = path.join(tmpDir.name, exampleId);
+      fs.rmSync(exampleOutputPath, { recursive: true, force: true });
+      fs.mkdirSync(exampleOutputPath, { recursive: true });
+      fs.cpSync(exampleTmpPath, exampleOutputPath, { recursive: true });
+    }
+    consola.success("Done");
+  }
+}
+
+const DELETE_NPM_RC: Mutator = {
+  filePattern: ".npmrc",
+  mutate: () => ({
+    type: "delete",
+  }),
+};
+
+const UPDATE_PACKAGE_JSON: Mutator = {
+  filePattern: "package.json",
+  mutate: (template, content) => ({
+    type: "modify",
+    newContent: content.replace(
+      "\"@osdk/examples.one.dot.one\": \"latest\"",
+      "\"@osdk/examples.one.dot.one\": \"workspace:*\"",
+    ).replace(
+      `"name": "${templateExampleId(template)}"`,
+      `"name": "@osdk/examples.${templateCanonicalId(template)}"`,
+    ),
+  }),
+};
+
+const UPDATE_README: Mutator = {
+  filePattern: "README.md",
+  mutate: (template) => ({
+    type: "modify",
+    newContent: readme(template),
+  }),
+};
+
+const MUTATORS: Mutator[] = [
+  DELETE_NPM_RC,
+  UPDATE_PACKAGE_JSON,
+  UPDATE_README,
+];
```

Comment:
Just a nit: can we move these under the interfaces as it makes reading the implementation for the first time easier without having to jump around few times to understand what is happening in the for loop